dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1141,6 +1158,40 @@ CDefaultContextMenu::InvokeRegVerb(
 
     PStaticShellEntry pEntry = &m_StaticEntries.GetAt(it);
 
+    WCHAR VerbKey[sizeof("shell\\") + MAX_VERB];
+    hr = StringCbPrintfW(VerbKey, sizeof(VerbKey), L"shell\\%s", pEntry->Verb.GetString());
+    HKEY hVerbKey;
+    if (SUCCEEDED(hr) && m_pDataObj &&
+        RegOpenKeyExW(pEntry->hkClass, VerbKey, 0, KEY_READ, &hVerbKey) == ERROR_SUCCESS)
+    {
+        DWORD KeyState = 0;
+        if (lpcmi->fMask & CMIC_MASK_SHIFT_DOWN)
+            KeyState |= MK_SHIFT;
+        if (lpcmi->fMask & CMIC_MASK_CONTROL_DOWN)
+            KeyState |= MK_CONTROL;
+        POINTL *pPtl = NULL;
+        C_ASSERT(sizeof(POINT) == sizeof(POINTL));
+        if (lpcmi->fMask & CMIC_MASK_PTINVOKE)
+            pPtl = (POINTL*)&lpcmi->ptInvoke;
+        CLSID clsid;
+
+        // TODO: IExecuteCommand
+
+        IDropTarget *pDT;
+        hr = SHELL_GetRegCLSID(hVerbKey, L"DropTarget", L"CLSID", clsid);
+        if (SUCCEEDED(hr))
+            hr = CoCreateInstance(clsid, NULL, CLSCTX_ALL, IID_PPV_ARG(IDropTarget, &pDT));
+        if (SUCCEEDED(hr))
+        {
+            IUnknown_SetSite(pDT, static_cast<IContextMenu*>(this));
+            hr = SHSimulateDrop(pDT, m_pDataObj, KeyState, pPtl, NULL);
+            IUnknown_SetSite(pDT, NULL);
+            pDT->Release();
+            return hr;
```

Comment:
What's happening with the hVerbKey registry key handle there?

Reply:
Good catch. I originally had the call to `SHELL_ErrorBox` here but had to back that out because it does not exist yet and we should handle all errors in a common place. I will PR that when `SHELL_ErrorBox` lands and I get around to checking what Windows does exactly in the various cases.